src/lib.rs

```diff
@@ -1168,43 +1171,55 @@ fn des3_adjust_parity_bits(mut key: Vec<u8>) -> Vec<u8> {
 ///
 /// # Returns
 ///
-/// The name of the digest algorithm if the OID is recognized, else an `EmrtdError`.
+/// The digest algorithm if the OID is recognized, else an `EmrtdError`.
 ///
 /// # Errors
 ///
 /// * `EmrtdError` if an unsupported OID is given.
 #[cfg(feature = "passive_auth")]
-fn oid2digestalg(oid: &rasn::types::ObjectIdentifier) -> Result<MessageDigest, EmrtdError> {
-    let digest_alg_oid_dict: [(&Oid, MessageDigest); 6] = [
-        (
-            Oid::const_new(&[2, 16, 840, 1, 101, 3, 4, 2, 4]),
-            MessageDigest::sha224(),
-        ),
-        (
-            Oid::const_new(&[2, 16, 840, 1, 101, 3, 4, 2, 3]),
-            MessageDigest::sha512(),
-        ),
-        (
-            Oid::const_new(&[2, 16, 840, 1, 101, 3, 4, 2, 2]),
-            MessageDigest::sha384(),
-        ),
-        (
-            Oid::const_new(&[2, 16, 840, 1, 101, 3, 4, 2, 1]),
-            MessageDigest::sha256(),
-        ),
-        (Oid::const_new(&[1, 3, 14, 3, 2, 26]), MessageDigest::sha1()),
-        (
-            Oid::const_new(&[1, 2, 840, 113549, 2, 5]),
-            MessageDigest::md5(),
-        ),
-    ];
-    for (digest_oid, digest) in digest_alg_oid_dict {
-        if oid.eq(digest_oid) {
-            return Ok(digest);
-        }
+fn oid2digestalg(oid: &rasn::types::ObjectIdentifier) -> Result<Box<dyn DynDigest>, EmrtdError> {
+    if Oid::const_new(&[2, 16, 840, 1, 101, 3, 4, 2, 4]).eq(oid) {
+        Ok(Box::new(sha2::Sha224::default()))
+    } else if Oid::const_new(&[2, 16, 840, 1, 101, 3, 4, 2, 3]).eq(oid) {
+        Ok(Box::new(sha2::Sha512::default()))
+    } else if Oid::const_new(&[2, 16, 840, 1, 101, 3, 4, 2, 2]).eq(oid) {
+        Ok(Box::new(sha2::Sha384::default()))
+    } else if Oid::const_new(&[2, 16, 840, 1, 101, 3, 4, 2, 1]).eq(oid) {
+        Ok(Box::new(sha2::Sha256::default()))
```

Comment:
There might be a better way. But probably slower
https://github.com/RustCrypto/formats/blob/0bd9dea37e3d8cccdb5f7ba5a402f62553d07e05/cms/src/builder.rs#L1061-L1077

Reply:
Hey, thanks for the comment. It seems that with that approach, I would need to use an oid database (`der::oid::db::DB`). Since oids do not change, I think it's better to have them hardcoded. Even though it looks cleaner, I would rather have very little dependency footprint. Currently, I have quite many dependencies but I plan to reduce them in time.